tests/webgpu_basic_rendering.cpp

```diff
@@ -365,6 +367,10 @@ int main() {
         run();
     });
 
-    // result will be reported when the main_loop completes
-    emscripten_exit_with_live_runtime();
+    // Result will be reported when the main_loop completes.
+    // emscripten_exit_with_live_runtime isn't needed because the WebGPU
+    // callbacks should all automatically keep the runtime alive until
+    // emscripten_set_main_loop, and that should keep it alive until
+    // emscripten_cancel_main_loop.
+    return 99; // Return code gets ignored if the runtime correctly stays alive.
```

Comment:
Aha, that makes sense, because visually the test is completing (it shows the triangle and prints the right text) so this isn't preventing the runtime from continuing.

Perhaps it's best to stick to the usual structure then, return 99 and later exit(0). I could have another test for implicit exits.

Reply:
Hm, perhaps another test for implicit exit isn't needed. If I forget a runtimeKeepalivePop, it hangs even with `exit(0)`. It was the `emscripten_force_exit(0)` that was hiding that error condition before, but I don't think I need that.